## Data preprocessing
#### Nov.9.2022

In [2]:
%%capture
import pandas as pd
import numpy as np

df_dietary = pd.read_sas('./dataset/P_DSQTOT.XPT')
df_nutrition = pd.read_sas('./dataset/P_DR1TOT.XPT')
df_dental = pd.read_sas('./dataset/P_OHXDEN.XPT')


In [3]:
df_data = pd.merge(df_dietary, df_nutrition, on='SEQN')
df_data = pd.merge(df_data, df_dental, on='SEQN')
df_data.info()
df_data.head()
df_data.to_csv('train_data.csv')

<class 'pandas.core.frame.DataFrame'>
Int64Index: 13772 entries, 0 to 13771
Columns: 375 entries, SEQN to OHX31SE
dtypes: float64(245), object(130)
memory usage: 39.5+ MB


In [4]:
df_dropna = df_data.dropna()
df_dropna.info()


<class 'pandas.core.frame.DataFrame'>
Int64Index: 0 entries
Columns: 375 entries, SEQN to OHX31SE
dtypes: float64(245), object(130)
memory usage: 0.0+ bytes


#### Summary: Our dataset has 13772 observations and 374 features with a lot of empty entries. If we drop na, there is simply nothing left.